examples/pipeline_wavernn/utils.py

```diff
@@ -0,0 +1,61 @@
+import logging
+import os
+import shutil
+from collections import defaultdict, deque
+
+import torch
+
+
+class MetricLogger:
+    r"""Logger for model metrics
+    """
+
+    def __init__(self, group, print_freq=1):
+        self.print_freq = print_freq
+        self._iter = 0
+        self.data = defaultdict(lambda: deque(maxlen=self.print_freq))
+        self.data["group"].append(group)
+
+    def __call__(self, key, value):
+        self.data[key].append(value)
+
+    def _get_last(self):
```

Comment:
Thinking out loud: Instead of `_get_last` we could add support for `metric_logger[:]` via `__getitem__` to return this. We could then add the snippet I mention [here](https://github.com/pytorch/audio/pull/749/files#r455140233).

Reply:
```python
class X:
  def __getitem__(self, key):
    if isinstance(key, slice) and key.start is None and key.stop is None and key.step is None:
      return {k: v[-1] for k, v in self.data.items()}
  return self.data[key][-1]
```
this means we could do:
```python
    def __str__(self):
        return str(self[:])
```

Reply:
however, this returns a dictionary, not the list of values. maybe this wouldn't be natural :)